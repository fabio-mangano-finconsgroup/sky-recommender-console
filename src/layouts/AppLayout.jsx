import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import Box from '@mui/material/Box';

/**
 * Main layout of the application.
 */

const AppLayout = ({ children }) => {
  return (
    <div data-testid="app-layout">
      <AppBar data-testid="app-bar" position="static">
        <Container maxWidth="xl">
          <Toolbar disableGutters>
            <Typography variant="h6" component="div">
              SKY Recommender Console
            </Typography>
          </Toolbar>
        </Container>
      </AppBar>
      <Box sx={{ m: 2 }} />
      <Container maxWidth="xl">{children}</Container>
    </div>
  );
};

AppLayout.propTypes = {
  /** Page component */
  children: PropTypes.node,
};

export default AppLayout;
