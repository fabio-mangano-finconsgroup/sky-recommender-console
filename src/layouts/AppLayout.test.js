import { render, screen } from '@testing-library/react';
import AppLayout from './AppLayout';

const DummyPage = () => <div>Page</div>;

describe('AppLayout', () => {
  it('should render', async () => {
    render(
      <AppLayout>
        <DummyPage />
      </AppLayout>,
    );

    await screen.findByText('SKY Recommender Console');

    expect(document.body.childNodes[0].children).toMatchSnapshot();
  });
});
