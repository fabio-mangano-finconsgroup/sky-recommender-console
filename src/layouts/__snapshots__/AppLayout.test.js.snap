// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout should render 1`] = `
HTMLCollection [
  <div
    data-testid="app-layout"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      data-testid="app-bar"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            SKY Recommender Console
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-1c5ij41"
    />
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div>
        Page
      </div>
    </div>
  </div>,
]
`;
